pkg/block/s3/inventory.go

```diff
@@ -23,35 +22,35 @@ type Manifest struct {
 	URL                string          `json:"-"`
 	InventoryBucketArn string          `json:"destinationBucket"`
 	SourceBucket       string          `json:"sourceBucket"`
-	Files              []inventoryFile `json:"files"` // inventory list files, each contains a list of objects
+	Files              []InventoryFile `json:"files"` // inventory list files, each contains a list of objects
 	Format             string          `json:"fileFormat"`
 	CreationTimestamp  string          `json:"creationTimestamp"`
 	inventoryBucket    string
 }
 
-type inventoryFile struct {
+type InventoryFile struct {
 	Key      string `json:"key"` // an s3 key for an inventory list file
 	firstKey string
 	lastKey  string
 }
 
 func (a *Adapter) GenerateInventory(ctx context.Context, logger logging.Logger, manifestURL string, shouldSort bool, prefixes []string) (block.Inventory, error) {
-	return GenerateInventory(logger, manifestURL, a.s3, s3inventory.NewReader(ctx, a.s3, logger), shouldSort, prefixes)
+	m, err := a.loadManifest(ctx, manifestURL)
```

Comment:
As discussed, we shall open an issue to adding import integration tests.

Reply:
Opened #2332

Reply:
Thanks!